src/Sentry/HubExtensions.cs

```diff
@@ -12,6 +12,58 @@ namespace Sentry
     [EditorBrowsable(EditorBrowsableState.Never)]
     public static class HubExtensions
     {
+        /// <summary>
+        /// Configures the scope and captures an event.
+        /// </summary>
+        public static SentryId CaptureEvent(this IHub hub, SentryEvent @event, Action<Scope> configureScope)
+        {
+            var id = SentryId.Empty;
+
+            // The callback is executed immediately, so the ID is set before returning
+            hub.WithScope(scope =>
+            {
+                configureScope(scope);
+                id = hub.CaptureEvent(@event, scope);
+            });
+
+            return id;
+        }
+
+        /// <summary>
+        /// Configures the scope and captures a message.
+        /// </summary>
+        public static SentryId CaptureMessage(this IHub hub, string message, Action<Scope> configureScope,
+            SentryLevel level = SentryLevel.Info)
+        {
+            var id = SentryId.Empty;
+
+            // The callback is executed immediately, so the ID is set before returning
+            hub.WithScope(scope =>
```

Comment:
I'm not sure I understand. This method doesn't call `Push`/`Pop`, it calls `WithScope` which is a black box. In a single-scope-hub it may always use the same instance.

Reply:
`WithScope` won't work in single hub. It does push and pop.